test/test_bitfield.cpp

```diff
@@ -355,3 +355,56 @@ TORRENT_TEST(find_last_clear_misc)
 	test4.clear_bit(89);
 	TEST_EQUAL(test4.find_last_clear(), 91);
 }
+
+TORRENT_TEST(not_initialized)
+{
+	// in some cases for example peer disconnect before receiving bitfield
+	// we try to check a not initialized bitfield
+	bitfield test1(0);
+	TEST_EQUAL(test1.none_set(), true);
+	TEST_EQUAL(test1.all_set(), false);
+	TEST_EQUAL(test1.size(), 0);
+	TEST_EQUAL(test1.num_words(), 0);
+	TEST_EQUAL(test1.empty(), true);
+	TEST_CHECK(test1.data() == nullptr);
+	TEST_CHECK(test1.bytes() == nullptr);
+	TEST_EQUAL(test1.count(), 0);
+	TEST_EQUAL(test1.find_first_set(), -1);
+	TEST_EQUAL(test1.find_last_clear(), -1);
+
+	test1.clear_all();
+	TEST_EQUAL(test1.size(), 0);
+
+	test1.clear();
+	TEST_EQUAL(test1.size(), 0);
+
+	test1.set_all();
+	TEST_EQUAL(test1.size(), 0);
+
+	// don't test the following methods because they are optimized and would be
+	// only called after initialization to skip the null check
+	// get_bit, clear_bit, set_bit
```

Comment:
more specifically, they aren't defined for empty sets. just like you can't use ``operator[]`` on an empty vector